### PR TITLE
Github action using cache

### PR DESCRIPTION
**Pull request by krol3**
Using cache with buildx, related the issue [323](https://github.com/aquasecurity/tracee/issues/323)

### CODE REVIEW

File: .github/workflows/build-test.yaml
Comment (itaysk) [inline .github/workflows/build-test.yaml:35]: this is unnecessary. we push the images from the release script

File: .github/workflows/build-test.yaml
Comment (itaysk) [inline .github/workflows/build-test.yaml:31]: correct me if I'm wrong but won't this cache only between builds on the same commit?
we want to be able to cache across commits

**Review by itaysk — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test.yaml
Comment (krol3) [reply .github/workflows/build-test.yaml:20]: It's only to validate .yaml format (empty spaces, etc), also if you want to check common misspellings in text files: codespell-project/actions-codespell.
I will remove this yaml-lint.

File: .github/workflows/build-test.yaml
Comment (krol3) [reply .github/workflows/build-test.yaml:30]: Using this settings, we are doing cache across commits,
```
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache,mode=max
```

File: .github/workflows/build-test.yaml
Comment (krol3) [reply .github/workflows/build-test.yaml]: I go back to the origin/main branch

**Comment by grantseltzer** [comment]
@itaysk Did you have further comment on this, and if not I believe this will be good to merge after @krol3 resolves conflict?

**Comment by krol3** [comment]
I need to test it with the new updates. Please don't merge yet.

**Comment by itaysk** [comment]
@grantseltzer issues weren't addressed. @krol3 no rush, thanks for working on this

File: .github/workflows/pr-ebpf.yaml
Comment (itaysk) [inline .github/workflows/pr-ebpf.yaml:27]: the build step is now missing in the workflow isn't it?

Comment (krol3) [reply .github/workflows/pr-ebpf.yaml:27]: Now, It's working
- First build time (2m 53s): https://github.com/krol3/tracee/actions/runs/541588005
- Second time using cache(1m 51s): https://github.com/krol3/tracee/actions/runs/541596189
The actually time in the job pr-ebpf (1m 45s)

Comment (itaysk) [reply .github/workflows/pr-ebpf.yaml:27]: is it actually building the executable thought? i don't think so since you removed this command, but maybe I'm missing something

**Comment by krol3** [comment]
@itaysk  using cache for the golang artifacts we gain some time in the pr-rules workflow.

image

**Comment by krol3** [comment]
@itaysk I rerun the tracee-ebpf workflow. I have this result using the flag DOCKER_BUILDKIT=1

image

**Comment by itaysk** [comment]
@krol3 isn't this missing the cache action? now the only change is to opt in to buildkit but how is the cache persisted between runs?

**Comment by rafaeldtinoco** [comment]
@krol3 I believe this is outdated based on our recent efforts to have a new builder, new PR workflows, etc. I'm marking this as a draft so you can work this through, or close it, if you would like (just so we can better filter the on going PRs we have).

**Comment by danielpacak** [comment]
I think it would be much easier to open a new PR once we are done with the repo restructure. We no longer split PR workflow to tracee-ebpf and tracee-rules.

**Pull request closed without merging**
